### PR TITLE
update setCellProps for hierCluster

### PR DESCRIPTION
**Pull request by congyu-lu**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/matrix.cells.js [hunk 1]
```diff
@@ -169,24 +169,6 @@ export function setGeneVariantCellProps(cell, tw, anno, value, s, t, self, width
 			order: -1,
 			entry: { key: value.class, label: cell.label, fill: cell.fill, order, dt: value.dt, origin: value.origin }
 		}
-	} else if (value.dt >= dtgeneexpression) {
```
Comment (xzhou82) [inline client/plots/matrix.cells.js:172]: regarding this improper use, i've added comments about dt usage in yesterday's branch but my commit was somehow erased.
i will add it to your branch

Comment (xzhou82) [reply client/plots/matrix.cells.js:172]: committed

File: client/plots/hierCluster.js [hunk 2]
```diff
@@ -238,13 +233,13 @@ export class HierCluster extends Matrix {
...
 		const [min, max] = [-absMax, absMax]
-		// what's purpose of assigning this.geneExpValues{}, to signal something to matrix code?
-		this.geneExpValues = { scale, min, max }
+		// what's purpose of assigning this.hierClusterValues{}, to signal something to matrix code?
+		this.hierClusterValues = { scale, min, max }
 	}
 
 	getValueColor(value) {
 		const zScoreCap = this.settings.hierCluster.zScoreCap
-		return this.geneExpValues.scale((value - -zScoreCap) / (zScoreCap * 2))
```
Comment (airenzp) [inline client/plots/hierCluster.js:247]: did you meant -- or +?

File: client/plots/matrix.cells.js (hunk 1 above)
Comment (congyu-lu) [reply client/plots/matrix.cells.js:172]: thanks fixed now

File: client/plots/hierCluster.js (hunk 2 above)
Comment (congyu-lu) [reply client/plots/hierCluster.js:247]: here means two minus, so we could set the range to be (-zScoreCap, zScoreCap)

File: client/plots/matrix.serieses.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ export function getSerieses(data) {
 			for (const [i, value] of values.entries()) {
 				const cell = Object.assign({ key, siblingCells }, cellTemplate)
 				cell.valueIndex = i
-				let cellProps = setCellProps[t.tw.term.type]
+				const cellProps = t.grp.type == 'hierCluster' ? setCellProps['hierCluster'] : setCellProps[t.tw.term.type]
```
Comment (xzhou82) [inline client/plots/matrix.serieses.js:75]: geneExpression is not defined in `setCellProps{}`

**Review by airenzp — DISMISSED** [review]
Thanks for the quick fixes! It works.

File: client/plots/matrix.serieses.js (hunk 3 above)
Comment (xzhou82) [reply client/plots/matrix.serieses.js:75]: @congyu-lu please address this, so adding a gene exp term via Variables button can work.

File: client/plots/matrix.cells.js [hunk 4]
```diff
@@ -204,6 +186,47 @@ export function setGeneVariantCellProps(cell, tw, anno, value, s, t, self, width
 	}
 }
 
+export function setHierClusterCellProps(cell, tw, anno, value, s, t, self, width, height, dx, dy, i) {
```
Comment (xzhou82) [inline client/plots/matrix.cells.js:189]: why consider mclass/class/origin here?

File: client/plots/matrix.serieses.js (hunk 3 above)
Comment (airenzp) [reply client/plots/matrix.serieses.js:75]: it is added now

**Review by xzhou82 — APPROVED** [review]
thanks for the fix.
adding gene exp to oncomatrix seems not right, can look into it next

**Pull request merged**